spark3/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -159,6 +164,11 @@ public WriteBuilder newWriteBuilder(LogicalWriteInfo info) {
   public void deleteWhere(Filter[] filters) {
     Expression deleteExpr = SparkFilters.convert(filters);
 
+    if (deleteExpr == Expressions.alwaysFalse()) {
+      LOG.info("Skipping the delete operation as the condition is always false");
+      return;
+    }
```

Comment:
This is probably just me, but I would rather this just be an "if { log } else { setup delete }" than an "if { log; return } setupDelete "

Reply:
Just trying to touch fewer lines and avoid extra indentation for the block below.

Reply:
I agree with both of you. Normally, I'd prefer if/else to an extra return. But in this case it is better to return because we don't want to touch all of the lines that actually perform the delete.